bevy_rapier2d/examples/player_movement.rs

```diff
@@ -0,0 +1,74 @@
+use bevy::prelude::*;
+use bevy_rapier2d::physics::{RapierConfiguration, RapierPhysicsPlugin, RigidBodyHandleComponent};
+use bevy_rapier2d::rapier::dynamics::{RigidBodyBuilder, RigidBodySet};
+use bevy_rapier2d::rapier::geometry::ColliderBuilder;
+use bevy_rapier2d::rapier::na::Vector2;
+
+fn main() {
+    App::build()
+        .add_resource(WindowDescriptor {
+            title: "Player Movement Example".to_string(),
+            width: 1000,
+            height: 1000,
+            ..Default::default()
+        })
+        .add_startup_system(spawn_player.system())
+        .add_system(player_movement.system())
+        .add_plugin(RapierPhysicsPlugin)
+        .add_plugins(DefaultPlugins)
+        .run();
+}
+
+// The float value is essentially a player movement speed factor.
+struct Player(f32);
+
+fn spawn_player(
+    mut commands: Commands,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+    mut rapier_config: ResMut<RapierConfiguration>,
+) {
+    // Set gravity to 0.0 and spawn camera.
+    rapier_config.gravity = Vector2::zeros();
+    commands.spawn(Camera2dComponents::default());
+
+    let sprite_size_x = 40.0;
+    let sprite_size_y = 40.0;
+
+    // Spawn entity with `Player` struct as a component for access in movement query.
+    // If `can_sleep` is not set to false, entity will not respond to input after idle time.
+    commands
+        .spawn(SpriteComponents {
+            material: materials.add(Color::rgb(0.0, 0.0, 0.0).into()),
+            sprite: Sprite::new(Vec2::new(sprite_size_x, sprite_size_y)),
+            ..Default::default()
+        })
+        .with(RigidBodyBuilder::new_dynamic().can_sleep(false))
```

Comment:
You don't actually need the `.can_sleep(false)` here. The problem you're experiencing is likely because you're setting linvel directly instead of using `set_linvel()`